### PR TITLE
InputContextManager.save(): split monitor #1129

### PR DESCRIPTION
**Pull request by jukzi**
https://github.com/eclipse-pde/eclipse.pde/issues/1129

### CODE REVIEW

**Comment by jukzi** [comment]
@iloveeclipse please test if that solves your issue.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java [hunk 1]
```diff
@@ -77,9 +79,11 @@ public void dispose() {
 	 * Saves dirty contexts.
 	 */
 	public void save(IProgressMonitor monitor) {
-		for (InputContext context : inputContexts.values()) {
+		Collection<InputContext> values = inputContexts.values();
+		SubMonitor subMon = SubMonitor.convert(monitor, values.size());
+		for (InputContext context : values) {
 			if (context.mustSave())
-				context.doSave(monitor);
+				context.doSave(subMon.newChild(1));
```
Comment (laeubi) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java:86]:
```suggestion
				context.doSave(subMon.split(1));
```

**Comment by jukzi** [comment]
@iloveeclipse ping

**Comment by iloveeclipse** [comment]
> ping

I can't reproduce that, so I can't say if the patch makes any difference.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java (hunk 1 above)
Comment (jukzi) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java:86]: split() would check for cancel. i don't know if that is safe at this place and don't think it's needed - never had a lag there that i would like to cancel.

**Comment by jukzi** [comment]
> I can't reproduce that, so I can't say if the patch makes any difference.

nevertheless that code looks wrong, i suggest to submit.

**Comment by iloveeclipse** [comment]
> nevertheless that code looks wrong, i suggest to submit.

If you think so, please. I don't want to look into monitor code if I'm not forced to.

**Pull request merged**